pypeman/msgstore.py

```diff
@@ -484,3 +507,20 @@ async def search(self, start=0, count=10, order_by='timestamp', start_dt=None, e
 
     async def total(self):
         return self._total
+
+    async def _delete_meta_file(self, id):
+        meta_fpath = (Path(self.base_path) / str(id)).with_suffix(".meta")
+        meta_fpath.unlink()
+
+    async def delete(self, id):
+        fpath = Path(self.base_path) / str(id)
+        if not fpath.exists():
+            raise IndexError
+
+        with open(fpath, "rb") as f:
```

Comment:
if it's a path object you can (should) write.

`with fpath.open("rb") as f:`

Reply:
done